### DISCUSSION TITLE
connect timeout doesn't work

### DISCUSSION THREAD

**Discussion by arch-user-france1** [prisma/prisma#17360] [Q&A] [answered]
### Bug description

?sslcert=cert.pem => when I add ?connect_timeout=7 it fails - P1001: Can't reach database server at

I'm using Nextjs... So when I remove that connect_timeout it works fine on my computer, however on vercel I get a timeout, so I tried increasing the connect_timeout which is, as stated in the documentation, defaulted to 5.

Even if I do 300 it immediately tells me the connection timed out. Doesn't even wait.
I basically do this: mysql://USER:PASSWORD@HOST/DATABASE?connect_timeout=7

Worth noting is that I'm using MariaDB...

And I also tried with timeout = 5

### Expected behavior

Don't time out immediately.

### Prisma information

-DATABASE_URL="mysql://myUser:mjasndjkasndmb@mhm.ldt:3306/myDB?sslcert=cert.pem?connect_timeout=5"

### Environment & setup

- OS: MacOS, Vercel systems
- Database: MariaDB
- Node.js version: 18, Vercel Node serverless function

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 4.8.1
@prisma/client          : 4.8.1
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Format Wasm             : @prisma/prisma-fmt-wasm 4.8.0-61.d6e67a83f971b175a593ccc12e15c4a757f93ffe
Default Engines Hash    : d6e67a83f971b175a593ccc12e15c4a757f93ffe
Studio                  : 0.479.0
```

**Comment by arch-user-france1** [comment, marked as answer]
Is the syntax correct here?
According to the documentation that should work..

**Reply by janpio** [reply]
No, the second one has to be connected with a `&` instead of `?`:
```
DATABASE_URL="mysql://myUser:mjasndjkasndmb@mhm.ldt:3306/myDB?sslcert=cert.pem&connect_timeout=5
```

**Reply by arch-user-france1** [reply]
Thanks, seems to work. Sorry for the disruption.

**Reply by janpio** [reply]
Common mistake to make. We want to add a warning about that: https://github.com/prisma/prisma/issues/16677

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
